libs/go/aealite/protocols/dialogue.go

```diff
@@ -0,0 +1,437 @@
+package protocols
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"errors"
+	"fmt"
+)
+
+type Address string
+type Performative []string
```

Comment:
why a `Performative` is an array of strings?

In the Python version, we used an enumeration class to represent a "set of performatives (i.e. type identifiers)". However, a variable of type "PerformativeX" can be thought of as a specific value of that set, so I'd say `Performative` should be a string. Thoughts?